plugins/ui/src/deephaven/ui/hooks/use_table_listener.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from functools import partial
+from typing import Callable, Literal
+
+from deephaven.table import Table
+from deephaven.table_listener import listen, TableUpdate, TableListener
+from deephaven.execution_context import make_user_exec_ctx, ExecutionContext
+
+from .use_effect import use_effect
+
+LockType: Literal["shared", "exclusive"]
+
+
+def listener_with_ctx(
+    exec_ctx: ExecutionContext,
+    listener: Callable[[TableUpdate, bool], None],
+    update: TableUpdate,
+    is_replay: bool,
+) -> None:
+    """
+    Call the listener within an execution context.
+
+    Args:
+        exec_ctx: ExecutionContext: The execution context to use.
+        listener: Callable[[TableUpdate, bool], None]: The listener to call.
+        update: TableUpdate: The update to pass to the listener.
+        is_replay: bool: Whether the update is a replay.
+    """
+    with exec_ctx:
+        listener(update, is_replay)
+
+
+def with_ctx(
+    listener: Callable[[TableUpdate, bool], None]
+) -> partial[[TableUpdate, bool], None]:
+    """
+    Wrap the listener in an execution context.
+
+    Args:
+        listener: The listener to wrap.
+
+    Returns:
+        partial[[TableUpdate, bool], None]: The wrapped listener.
+    """
+    return partial(listener_with_ctx, make_user_exec_ctx(), listener)
```

Comment:
unless you're passing args to it, looks like this should be `get_exec_ctx()` instead (suggestion for demonstration purposes, import has to be adjusted too)
```suggestion
    return partial(listener_with_ctx, get_exec_ctx(), listener)
```